hnn_core/network.py

```diff
@@ -25,7 +25,7 @@
 from .check import _check_gids, _gid_to_type, _string_input_to_list
 
 
-def _create_cell_coords(n_pyr_x, n_pyr_y, zdiff, inplane_distance):
+def _create_cell_coords(n_pyr_x, n_pyr_y, zdiff, inplane_distance, cell_names):
```

Comment:
I'd recommend renaming `cell_names` -> `cell_types` for consistency, which will be the `Network.cell_types` attribute containing all the cell names + cell templates (the latter is important too!).

Then, the keys of `pos_dict` will be made to exactly match those of `cell_types`. Do you see where I'm going with this?